include/libmpu/mpu6050.hpp

```diff
@@ -1,5 +1,201 @@
 #pragma once
 
-namespace hal::mpu {
+#include <libhal-util/bit.hpp>
+#include <libhal-util/i2c.hpp>
+#include <libhal-util/map.hpp>
+#include <libhal-util/serial.hpp>
+#include <libhal/accelerometer.hpp>
+
+namespace hal::mpu
+{
+class mpu6050 : public hal::accelerometer
+{
+ public:
+  /// The device address when A0 is connected to GND.
+  static constexpr hal::byte address_ground = 0b110'1000;
+  /// The device address when A0 is connected to V+.
+  static constexpr hal::byte address_voltage_high = 0b110'1001;
+  /// The device address when A0 is connected to SDA.
+  static constexpr hal::byte address_sda = 0b100'1010;
+  /// The device address when A0 is connected to SCL.
+  static constexpr hal::byte address_scl = 0b100'1011;
+
+  /// The address of the read-only register containing the temperature data.
+  static constexpr hal::byte xyz_register = 0x3B;
+  /// The address of the register used to configure gravity scale the device.
+  static constexpr hal::byte configuration_register = 0x1C;
+  /// The address of the register used to initilize the device.
+  static constexpr hal::byte initalizing_register = 0x6B;
+
+  /// The command to enable one-shot shutdown mode.
+  static constexpr hal::byte who_am_i = 0x75;
+
+  static result<mpu6050> create(hal::i2c & i2c,
+                                hal::byte device_address = address_ground)
+  {
+    mpu6050 mpu(i2c, device_address);
+    HAL_CHECK(mpu.is_valid_device());
+    HAL_CHECK(mpu.active_mode(true));
+    return mpu;
+  }
+
+  hal::status configure_full_scale(uint32_t gravity_scale)
+  {
+    hal::byte gravity_code;
+
+    switch (gravity_scale)
+    {
+      case 2:
+        gravity_code = 0x00;
+        m_gscale     = 2.0f;
+        break;
+      case 4:
+        gravity_code = 0x01;
+        m_gscale     = 4.0f;
+        break;
+      case 8:
+        gravity_code = 0x02;
+        m_gscale     = 8.0f;
+        break;
+      case 16:
+        gravity_code = 0x03;
+        m_gscale     = 16.0f;
+        break;
+      default: return hal::new_error(std::errc::illegal_byte_sequence);
+    }
+
+    // Write in the full scale range; but leave the self test and high pass
+    // filter config untouched
+    constexpr auto scale_mask = hal::bit::mask::from<3, 4>();
+
+    auto config =
+        HAL_CHECK(hal::write_then_read<1>(*m_i2c,
+                                          m_address,
+                                          std::array{ configuration_register },
+                                          hal::never_timeout()));
+
+    // config = bit::Insert(config, gravity_code, kScaleMask);
+
+    hal::bit::modify(config[0]).insert<scale_mask>(gravity_code);
+    HAL_CHECK(hal::write(*m_i2c,
+                         m_address,
+                         std::array{ configuration_register, config[0] },
+                         hal::never_timeout()));
+
+    return hal::success();
```

Comment:
```suggestion
    return hal::write(*m_i2c,
                         m_address,
                         std::array{ configuration_register, config[0] },
                         hal::never_timeout());
```

Might as well remove the extra check if you can.